renku/cli/_checks/migrate_datasets.py

```diff
@@ -52,3 +76,22 @@ def check_missing_files(client):
         )
 
     return False
+
+
+def check_dataset_resources(client):
+    """Find missing datasets or files listed in metadata."""
+    missing_datasets = defaultdict(list)
+    missing_files = defaultdict(list)
+
+    for path, dataset in client.datasets.items():
```

Comment:
it might be better to load the `yaml` or `json` directly - then you can also check if the file paths make sense, e.g. are valid inside the repository.

Reply:
What do you mean? `client.datasets` will load it directly and since this is validation and it will do just that `check if paths make sense`.